core/src/model/document_type.rs

```diff
@@ -0,0 +1,17 @@
+#[derive(Debug)]
+pub enum DocumentType {
+    Text,
+    Drawing,
+    Other,
+}
+
+// todo: be more exhaustive
+impl DocumentType {
+    pub fn from_file_name_using_extension(name: &str) -> DocumentType {
+        match name.split('.').last() {
+            Some("md") => DocumentType::Text,
```

Comment:
criteria could be trivially broader, include `txt` and `markdown` for starters. Somewhere in the codebase there already is a list of these that include `rs` for instance.